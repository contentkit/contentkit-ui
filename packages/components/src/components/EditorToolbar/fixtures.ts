export const anchorOrigin = {
  vertical: 'bottom',
  horizontal: 'center',
}

export const transformOrigin = {
  vertical: 'top',
  horizontal: 'center'
}