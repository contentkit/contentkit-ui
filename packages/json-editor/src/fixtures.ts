export enum Character {
  OPEN_SQUARE_BRACKET = '[',
  CLOSE_SQUARE_BRACKET = ']',
  OPEN_CURLY_BRACKET = '{',
  CLOSE_CURLY_BRACKET = '}',
  DOUBLE_QUOTE = '"'
}

export const CHARACTER_PAIRS = {
  [Character.OPEN_SQUARE_BRACKET]: Character.CLOSE_SQUARE_BRACKET,
  [Character.OPEN_CURLY_BRACKET]: Character.CLOSE_CURLY_BRACKET,
  [Character.DOUBLE_QUOTE]: Character.DOUBLE_QUOTE
}

export const OPEN_SCOPE_DELIMETERS = Object.keys(CHARACTER_PAIRS)
