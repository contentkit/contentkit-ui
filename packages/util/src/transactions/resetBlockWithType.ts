import setBlockProperties from './setBlockProperties'
import { Block } from '../fixtures'
import { List } from 'immutable'
import { EditorState } from 'draft-js'

function resetBlockWithType (editorState: EditorState, newType : string = Block.UNSTYLED) {
  return setBlockProperties(editorState, {
    type: newType,
    text: '',
    characterList: List()
  })
}

export default resetBlockWithType
