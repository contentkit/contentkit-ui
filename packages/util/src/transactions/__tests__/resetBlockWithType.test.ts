import resetBlockWithType from '../resetBlockWithType'
import { EditorState, SelectionState, ContentState, ContentBlock, CharacterMetadata } from 'draft-js'
import { List, Repeat } from 'immutable'
import { Block } from '../../fixtures'

const getEditorState = (blockArray, focusKey) => {
  const editorState = EditorState.createWithContent(
    ContentState.createFromBlockArray(blockArray)
  )
  return EditorState.acceptSelection(editorState,
    new SelectionState({
      anchorKey: focusKey,
      anchorOffset: 1,
      focusKey: focusKey,
      focusOffset: 1,
      isBackward: false
    })
  )
}

it('resetBlockWithType should reset the current block to the specified type and set the text to ""', () => {
  const blocksArray = [
    new ContentBlock({ key: 'A', type: Block.BLOCKQUOTE }),
    new ContentBlock({
      key: 'B',
      type: Block.UL,
      text: 'foo',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({
      key: 'C',
      type: Block.UL,
      text: 'bar',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    })
  ]

  const editor = getEditorState(blocksArray, 'C')

  expect(
    resetBlockWithType(editor, Block.UNSTYLED).getCurrentContent().getBlockMap()
  ).toMatchSnapshot()
})
