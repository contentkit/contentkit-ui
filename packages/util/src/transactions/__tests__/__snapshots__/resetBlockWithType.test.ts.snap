// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resetBlockWithType should reset the current block to the specified type and set the text to "" 1`] = `
Immutable.OrderedMap {
  "A": Immutable.Record {
    "key": "A",
    "type": "blockquote",
    "text": "",
    "characterList": Immutable.List [],
    "depth": 0,
    "data": Immutable.Map {},
  },
  "B": Immutable.Record {
    "key": "B",
    "type": "unordered-list-item",
    "text": "foo",
    "characterList": Immutable.List [
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
      Immutable.Record {
        "style": Immutable.OrderedSet [],
        "entity": null,
      },
    ],
    "depth": 0,
    "data": Immutable.Map {},
  },
  "C": Immutable.Record {
    "key": "C",
    "type": "unstyled",
    "text": "",
    "characterList": Immutable.List [],
    "depth": 0,
    "data": Immutable.Map {},
  },
}
`;
